src/components/FileInput/FileInput.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { ChangeEventHandler, useLayoutEffect, useRef } from "react";
+const fileInput = require("@uswds/uswds/packages/usa-file-input/src");
+
+type IntrinsicElements = JSX.IntrinsicElements["input"];
+
+interface Props extends IntrinsicElements {
+  id: string;
+  name: string;
+  multipleFiles?: boolean;
+  label?: string;
+  acceptedFileTypes?: string[];
+  hintText?: string;
+  disabled?: boolean;
+  error?: boolean;
+  errorMessage?: string;
+  onChange?: ChangeEventHandler<HTMLInputElement>;
+}
+
+/**
+ * **FileInput Component**
+ *
+ * @param {string}    id                  The id to uniquely identify the file input.
+ * @param {string}    name                Name of the file input field.
+ * @param {boolean}   multipleFiles       Controls if the input accepts multiple files.
+ * @param {string}    label               Used to label the file input in the UI.
+ * @param {string[]}  acceptedFileTypes   Array of string file types accepeted in this file input.
+ * @param {string}    hintText            Text to show to the user as a guide of the type of accepateble files.
+ * @param {boolean}   disabled            Controls whether the file input is disabled.
+ * @param {boolean}   error               Triggers error message and error styling.
+ * @param {string}    errorMessage        Message to show when an error is present.
+ * @param {string}    onChange            Function that is called when the File Input changes.
+ *
+ */
+
+export const FileInput: React.FC<Props> = ({
+  id,
+  multipleFiles = false,
+  label,
+  name,
+  error = false,
+  errorMessage,
+  acceptedFileTypes = [
+    ".pdf",
+    ".doc",
+    ".docx",
+    ".xlsx",
+    ".jpg",
+    ".jpeg",
+    ".png",
+  ],
+  hintText,
+  onChange,
+  ...rest
+}) => {
+  const fileInputRef = useRef<HTMLSpanElement>(null);
+  useLayoutEffect(() => {
+    const tooltipElement = fileInputRef.current;
+    // Safeguard to turn off JS and turn back on if hot reloaded
+    if (typeof fileInput.off === "function") fileInput.off(tooltipElement);
+    if (typeof fileInput.on === "function") fileInput.on(tooltipElement);
+
+    return () => {
+      if (typeof fileInput.off === "function") fileInput.off(tooltipElement);
+    };
+  });
```

Comment:
Not quite sure how to test this but the coverage test doesn't like this

Reply:
We might be able to mock the import and intentionally remove those functions. Not the absolute highest priority IMO.